### PR TITLE
Add types for padding.PSS class.

### PR DESCRIPTION
**Pull request by jonathanslenders**
This adds types to the ``PSS`` class, similar to the types that are present in typeshed:

https://github.com/python/typeshed/blob/04100b9cec9a5a25217e5e5d45e77560ef3ced48/stubs/cryptography/cryptography/hazmat/primitives/asymmetric/padding.pyi#L25

### CODE REVIEW

**Comment by alex** [comment]
Thanks for the PR! Unfortunately as you can see, mypy is unhappy :-( These challenges are why we haven't typed these yet. If you can find a way to make everything happy that'd be great!

File: tests/hazmat/backends/test_openssl.py [hunk 1]
```diff
@@ -410,7 +411,7 @@ def test_rsa_padding_unsupported_mgf(self):
         assert (
             backend.rsa_padding_supported(
                 padding.OAEP(
-                    mgf=DummyMGF(),  # type: ignore[arg-type]
+                    mgf=cast(padding.MGF1, DummyMGF()),
```
[the remark this answers is not in the document]
Comment (alex) [reply tests/hazmat/backends/test_openssl.py:414]: MGF is an A+ name.

Thanks!

[the remark this answers is not in the document]
Comment (jonathanslenders) [reply tests/hazmat/backends/test_openssl.py:414]: Two questions:
- Can I use ``typing.Protocol`` or is ``abc.ABC`` preferred?
- The interface needs to define an ``_algorithm: HashAlgorithm`` field. Which hash algorithm should the ``DummyMGF`` use in this case?

[the remark this answers is not in the document]
Comment (alex) [reply tests/hazmat/backends/test_openssl.py]: 1. abc.ABC is currently what we use for our interfaces, let's stick with that
2. literally anything :-) sha256 is fine

**Comment by jonathanslenders** [comment]
@alex: I'm ready with this one. Hope this is okay. ;)

**Review by reaperhulk — APPROVED** [review]

**Pull request merged**